oauthproxy.go

```diff
@@ -955,6 +963,52 @@ func (p *OAuthProxy) getAuthenticatedSession(rw http.ResponseWriter, req *http.R
 	return session, nil
 }
 
+func (p *OAuthProxy) validateGroups(groups []string) bool {
+	if len(p.AllowedGroups) == 0 {
+		return true
+	}
+
+	return checkGroupMembership(groups, buildGroupSet(p.AllowedGroups))
+}
```

Comment:
Does this, paired with #797 not mean that we are checking group membership now in multiple locations? Is that desirable?

Reply:
Yeah, I think I tagged this as WIP waiting for #797 -- I started work on them both at the same time, but this will need to adapt to the new architecture that is setup there.